jvm-agent/src/main/org/jetbrains/lincheck/jvm/agent/transformers/MethodCallTransformerBase.kt

```diff
@@ -39,8 +40,8 @@ internal abstract class MethodCallTransformerBase(
     override val requiresOwnerNameAnalyzer: Boolean = true
 
     override fun visitMethodInsn(opcode: Int, owner: String, name: String, desc: String, itf: Boolean) = adapter.run {
-        // TODO: do not ignore <init>
-        if (name == "<init>" || isIgnoredMethod(className = owner)) {
+        // TODO: JBRes-6844 do not ignore <init> for lincheck managed strategy
+        if ((name == "<init>" && !isInTraceRecorderMode) || isIgnoredMethod(className = owner)) {
```

Comment:
Could you please instead add a new flag to `TransformationConfiguration`, like `trackConstructorCalls`. 

It is better to have single entry point for configuring transformations (`TransformationConfiguration` class) and not spread the accesses to global properties like `isInTraceRecorderMode`.